### PR TITLE
Prepare GORM lib for v1 -> v2 migration

### PR DESCRIPTION
**Pull request by peterdeme**
Two changes:
- GORM V2 does not have `RecordNotFound()` method which would be one of the biggest changes in our `backend` repo.
- `DB()` method returns `(*sql.DB, error)` in GORM v2, so I modified that too.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -508,10 +521,11 @@ func (s *DB) Exec(sql string, values ...interface{}) *DB {
...
 // Model specify the model you would like to run db operations
-//    // update all users's name to `hello`
-//    db.Model(&User{}).Update("name", "hello")
-//    // if user's primary key is non-blank, will use it as condition, then will only update the user's name to `hello`
-//    db.Model(&user).Update("name", "hello")
+//
+//	// update all users's name to `hello`
+//	db.Model(&User{}).Update("name", "hello")
+//	// if user's primary key is non-blank, will use it as condition, then will only update the user's name to `hello`
+//	db.Model(&user).Update("name", "hello")
```
Comment (peterdeme) [inline main.go:528]: all this stuff got autoformatted by my IDE sorry 😅

File: main.go [hunk 2]
```diff
@@ -624,16 +638,6 @@ func (s *DB) NewRecord(value interface{}) bool {
 	return s.NewScope(value).PrimaryKeyZero()
 }
 
-// RecordNotFound check if returning ErrRecordNotFound error
-func (s *DB) RecordNotFound() bool {
```
Comment (peterdeme) [inline main.go:628]: This is change `#1`

File: main.go [hunk 3]
```diff
@@ -126,12 +128,15 @@ func (s *DB) Close() error {
 
 // DB get `*sql.DB` from current connection
 // If the underlying database connection is not a *sql.DB, returns nil
-func (s *DB) DB() *sql.DB {
+//
+// Note: originally this panics in GORM v1 is !ok, but we prepare for a V2
+// migration so we return an error as well.
+func (s *DB) DB() (*sql.DB, error) {
```
Comment (peterdeme) [inline main.go:134]: this is change `#2`

File: main.go [hunk 4]
```diff
@@ -126,12 +128,15 @@ func (s *DB) Close() error {
 
 // DB get `*sql.DB` from current connection
 // If the underlying database connection is not a *sql.DB, returns nil
-func (s *DB) DB() *sql.DB {
+//
+// Note: originally this panics in GORM v1 if !ok, but we prepare for a V2
+// migration so we return an error here instead.
+func (s *DB) DB() (*sql.DB, error) {
```
Comment (peterdeme) [inline main.go:134]: this is change `#2`

File: main.go (hunk 2 above)
Comment (TheMacies) [reply main.go:628]: Maybe, for easier backwards comp, we could supply a new implementation based on the new `errors.Is` approach. Something like 
```
func (s *DB) RecordNotFound() bool {
return errors.Is(s.Error, gorm.ErrRecordNotFound)
}
```

Comment (peterdeme) [reply main.go:628]: Sure but I don't want backward compatibility at all, I want to get rid of the fork.

Comment (TheMacies) [reply main.go:628]: oh, I thought this fork was adding some value to the original Gorm, but it in fact it does not. Nevermind me then, let's kill the fork.

Comment (peterdeme) [reply main.go:628]: Initially, it did add some value (@michalg9 had a bugfix), but then the original author completely rewrote GORM from scratch. Michal's bugfix is not relevant anymore.

File: main.go [hunk 5]
```diff
@@ -624,16 +638,6 @@ func (s *DB) NewRecord(value interface{}) bool {
 	return s.NewScope(value).PrimaryKeyZero()
 }
 
-// RecordNotFound check if returning ErrRecordNotFound error
-func (s *DB) RecordNotFound() bool {
-	for _, err := range s.GetErrors() {
```
Comment (TheMacies) [inline main.go:629]: A sanity check:

It seems this method was going through all the errors while what we do now is 
errors.Is(s.Error ....). Do we actually miss checking some errors or is errors.Is doing the job for us?

File: main.go (hunk 4 above)
Comment (TheMacies) [reply main.go:134]: <3

File: main.go (hunk 2 above)
Comment (michalg9) [reply main.go:628]: It did add value. The bugfix was about propagating nested preload errors (so e.g. "Stack.Account") to the top (basically if there was an error in such preload, we would not receive it from the query). I can imagine in gormv2 this was fixed, and if we do not have to use the fork anymore, let's not :D

File: main.go (hunk 5 above)
Comment (peterdeme) [reply main.go:629]: There is this helper method as well: https://github.com/spacelift-io/gorm/blob/447d578628011308498d9316838f59f93834967c/errors.go#L25

but yeah, all the unit tests are passing nicely, so it should work fine in this lib.

**Review by TheMacies — APPROVED** [review]

**Pull request merged**